tests/tests.rs

```diff
@@ -2844,49 +2844,53 @@ fn rotate_right_signed() {
 
 #[cfg(feature = "step_trait")]
 #[test]
-fn range_agrees_with_underlying() {
-    compare_range_32(u19::MIN, u19::MAX);
-    compare_range_64(u37::new(95_993), u37::new(1_994_910));
-    compare_range_128(u68::new(58_858_348), u68::new(58_860_000));
-    compare_range_128(u122::new(111_222_333_444), u122::new(111_222_444_555));
-    compare_range_32(u23::MIN, u23::MAX);
-    compare_range_64(u48::new(999_444), u48::new(1_005_000));
-    compare_range_128(u99::new(12345), u99::new(54321));
-
-    // with the `hint` feature enabled, ::value only exist with primitive types, not on all
-    // implementations. This makes some copy & paste necessary here.
```

Comment:
Because the `Number`/`SignedNumber` traits both provide a `value()` method (even when `hint` is enabled), adding that as a trait bound made it possible to get ride of the duplicate functions. Tested with both `hint` and `step_trait` enabled at the same time.

Reply:
General question: Should we introduce an ArbitraryNumber base trait that both Number and SignedNumber derive from?

(Of course, it would be nicer to name then Number, UnsignedNumber, SignedNumber, but that would be too breaking of a change :( )

Reply:
Yes! Though sadly I don't think it helps much here, we can't provide a blanket implementation due to the orphan rule. We'd likely want to implement `Step` for a signed/unsigned marker trait anyways to improve the quality of error messages.